cfpb/consumerfinance.gov
# Form explainers: color updates

The default design system link color had accessibility issues when displayed on the existing form explainer expandable background colors. This PR updates the explainer colors to improve accessibility, adds two additional colorways so the Wagtail explainer module can support up to 4 categories of annotations, and fixes the way the explainer Javascript is generated and added to the explainer template to resolve missing file errors on Content.

The form explainer colors are based on the CFPB design system palette but for the most part don't overlap with values in brand-colors.less (overlapping values are noted in comments in form-explainer.less).

---



## Additions

-  New colors for the explainers
- index.js file to init the explainer js in browser
- Full width text option in explainer content streamfield



## Changes

- Adds the explainer js to the page via the explainer template rather than the explainer block


## How to test this PR

1. Check out branch and run migrations and yarn
2. Create a FormExplainer page in Wagtail admin and verify that it is possible to add a Full Width Text block to the page
3. Add an explainer block to the page with 4 categories of annotations and verify that
    - the annotations have the expected background colors 
    - the explainer js is loaded / interactions work as expected 
    - WAVE tool does not report accessibility errors for links in the explainer expandables

## Screenshots

### Explainer colors
![explainer](https://user-images.githubusercontent.com/778171/176735556-fc3d3308-b047-45ed-ba62-fd1428e6f07a.png)

### Explainer admin
<img width="1321" alt="Screen Shot 2022-06-30 at 10 03 22 AM" src="https://user-images.githubusercontent.com/778171/176735997-abc09529-d282-4320-82ad-a2ecdf18cdaf.png">




## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance
